src/Microsoft.DotNet.Build.Tasks/DownloadFileFromUrl.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class DownloadFileFromUrl : BuildTask
+    {
+        
+        /// <summary>
+        /// The url to download the file from.
+        /// </summary>
+        [Required]
+        public string DownloadSource { get; set;}
+        
+        /// <summary>
+        /// The directory where the file will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationDirectory { get; set; }
+
+        /// <summary>
+        /// The file where the source will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationFileName { get; set; }
+
+        /// <summary>
+        /// The default is to emit warnings and not fail, the caller can fail if output parameter DownloadedFile is empty.
+        /// Set this to true to fail instead of just warn.
+        /// </summary>
+        public bool FailOnError { get; set; }
+
+        /// <summary>
+        /// The file that was downloaded, if there is an error this will be empty.
+        /// </summary>
+        [Output]
+        public string DownloadedFile { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                Log.LogMessage(MessageImportance.High, $"Downloading {DownloadSource} -> {DestinationFileName}");
+                Directory.CreateDirectory(DestinationDirectory);
+                string destinationPath = Path.Combine(DestinationDirectory, DestinationFileName);
+                using (FileStream stream = File.Create(destinationPath))
+                {
+                    using (var client = new HttpClient())
+                    {
+                        using (var result = client.GetAsync(DownloadSource).GetAwaiter().GetResult())
```

Comment:
please include some logic for handling the proxy. See https://github.com/dotnet/corefx/pull/23467 as example.

Reply:
I've fixed this. Talked to @davidsh and there is a difference on how this is done in .NET Core vs Desktop. The way to do it in .NET Core is not available until .NET4.7.1 so I had to use if def.